src/modules/Gapi.ts

```diff
@@ -0,0 +1,75 @@
+declare global {
+  interface Window {
+    gapi: any; 
+  }
+}
+
+const {
+    REACT_APP_GOOGLE_OAUTH_CLIENT_ID,
+    REACT_APP_GOOGLE_YOUTUBE_API_KEY,
+    REACT_APP_GOOGLE_AUTH_SCOPE_URL,
+    REACT_APP_GOOGLE_YOUTUBE_API_URL
+} = process.env;
+
+class GoogleApi {
+    
+    gapi: any;
+
+    constructor() {
+        const { gapi } = window;
+        this.gapi = gapi;
+    }
+
+    loadGapiClientAuth2 = async () => {
```

Comment:
You can use the method syntax for all these functions as follows:
```suggestion
    async loadGapiClientAuth2() {
```

Reply:
I will be using `google-api-nodejs-client` as suggested. And I will change this syntax.

Reply:
`loadGapiClientAuth2 = async () => {` Arrow function syntax required for proper `this` context inside the functions.

Reply:
I don't think that is the case. can you check again.